_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+| IRONIC_IMAGE | quay.io/metal3-io/ironic |
+| DOCKER_REGISTRY_IMAGE | docker.io/registry:latest |
+
+<br>
+
+> **Information:** In case it is expected to run self-modified container images, just overwrite the previous variables to match the right location. It is worth mentioning that a the container registry running as a privileged container in the host, so you can just upload there.
+
+<br>
+At this point, an Ansible role is run locally in order to complete the local configuration.
+
+```sh 
+ansible-playbook \
+  -e "working_dir=$WORKING_DIR" \
+  -e "virthost=$HOSTNAME" \
+  -i vm-setup/inventory.ini \
+  -b -vvv vm-setup/install-package-playbook.yml
+  ```
+
+This playbook imports two roles. One called `packages_installation`, which is in charge of installing a few more packages. The list of packages installed are listed as default Ansible variables [here](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/packages_installation/defaults/main.yml). The other role is based on a the [fubarhouse.golang](https://galaxy.ansible.com/fubarhouse/golang) Ansible Galaxy role. It is in charge of installing and configuring the exact `golang` version `1.12.12` defined in an Ansible variable [here](https://github.com/metal3-io/metal3-dev-env/blob/9fa752b90ed58fdadcd52c246d3023766dfcb2dc/vm-setup/install-package-playbook.yml#L12)
+
+Once the playbook is finished, a pod called `ironic-pod` is created. Inside that pod, a *privileged* `ironic-ipa-downloader` container is started and attached to the host network. This container is in charge of downloading the [Ironic Python Agent](https://docs.openstack.org/ironic-python-agent/latest/) (IPA) files to a shared volume defined by `IRONIC_IMAGE_DIR`. This folder is exposed by the `ironic` container through HTTP. 
+
+<br>
+> **Information:** The [Ironic Python Agent](https://docs.openstack.org/ironic-python-agent/latest/) is an agent for controlling and deploying Ironic controlled baremetal nodes. Typically run in a ramdisk, the agent exposes a REST API for provisioning servers.
+<br>
+<br>
+
+See below the code snippet that fullfill the task:
+
+```sh
+sudo podman run -d --net host --privileged --name ipa-downloader --pod ironic-pod -e IPA_BASEURI= -v /opt/metal3-dev-env/ironic:/shared quay.io/metal3-io/ironic-ipa-downloader /usr/local/bin/get-resource.sh
+```
+
+Below, it is shown the status of the pods and containers at this point:
+
+```sh
+[root@eko1 metal3-dev-env]# podman pod list --ctr-names
+POD ID         NAME         STATUS    CREATED      CONTAINER INFO                                             INFRA ID
+5a0d475351aa   ironic-pod   Running   6 days ago   [5a0d475351aa-infra] [ipa-downloader]                      18f3a8f61407
+```
+
+The process will wait until the `ironic-python-agent` (IPA) initramfs, kernel and headers file are downloaded successfully. See below the files downloaded along with the `CentOS 7` cloud image:
```

Comment:
The process will wait until the `ironic-python-agent` (IPA) initramfs, kernel and headers file**s** are downloaded successfully. Below the files downloaded along with the `CentOS 7` cloud image can be found: